EKF/ekf_helper.cpp

```diff
@@ -1215,9 +1215,17 @@ void Ekf::zeroCols(float (&cov_mat)[_k_num_states][_k_num_states], uint8_t first
 
 void Ekf::zeroOffDiag(float (&cov_mat)[_k_num_states][_k_num_states], uint8_t first, uint8_t last)
 {
-	// store diagonals
+	if (first > last) {
+		return;
+	}
+
+	// create dynamically sized array to save diagonal elemnets
+	const uint8_t array_size = last-first+1;
+	float * variances;
+	variances = new float [array_size];
```

Comment:
You'd need to delete variances when you're done with it. How about stack allocating `float variances[_k_num_states]` instead? I think a bit of wasted stack is better than a dynamic allocation.